bin/spark-shell

```diff
@@ -29,7 +29,7 @@ esac
 set -o posix
 
 if [ -z "${SPARK_HOME}" ]; then
-  export SPARK_HOME="$(cd "`dirname "$0"`"/..; pwd)"
+  source `dirname $0`/find-spark-home
```

Comment:
So for pip installed PySpark we need to do a bit more work to find the SPARK_HOME, using a common script to determine if we need to do more hunting rather than duplicating the logic.